src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -11,15 +10,17 @@
 import java.io.IOException;
 
 public class ProductListPageServlet extends HttpServlet {
-    private ProductDao productDao;
+    private ProductService productService;
+
     @Override
     public void init(ServletConfig config) throws ServletException {
         super.init(config);
-        productDao = new ArrayListProductDao();
+        productService = ProductServiceImpl.getInstance();
     }
+
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        request.setAttribute("products", productDao.findProducts());
+        request.setAttribute("products", productService.getProductDao().findProducts());
```

Comment:
Service should call dao, but not in this way
all dao methods should be present in service
service#findProducts() calls in method body dao.findProducts(), and for all other methods same logic is applied